### PR TITLE
JDK-8263256: Test java/net/Inet6Address/serialize/Inet6AddressSerializationTest.java fails due to dynamic reconfigurations of network interface during test

### PR DESCRIPTION
**Pull request by msheppar**
Test java/net/Inet6Address/serialize/Inet6AddressSerializationTest.java fails fails due to dynamic reconfigurations of network interface during test - this fix uses the utility method NetworkConfiguration::isSameInterface to replace NetworkInterface::equals method call. Additionally, for macosx the awdl and llw interfaces are excluded from the test

Please oblige and review this change to address issue https://bugs.openjdk.org/browse/JDK-8263256

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8263256](https://bugs.openjdk.org/browse/JDK-8263256): Test java/net/Inet6Address/serialize/Inet6AddressSerializationTest.java fails due to dynamic reconfigurations of network interface during test (**Bug** - P3)

### Reviewers
 * [Jaikiran Pai](https://openjdk.org/census#jpai) (@jaikiran - **Reviewer**) ⚠️ Review applies to [04c31fe5](https://git.openjdk.org/jdk/pull/16998/files/04c31fe58eae401ac222e8167828c95683c41d4a)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16998/head:pull/16998` \
`$ git checkout pull/16998`

Update a local copy of the PR: \
`$ git checkout pull/16998` \
`$ git pull https://git.openjdk.org/jdk.git pull/16998/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16998`

View PR using the GUI difftool: \
`$ git pr show -t 16998`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16998.diff">https://git.openjdk.org/jdk/pull/16998.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16998#issuecomment-1843266832)

### CODE REVIEW

File: test/jdk/java/net/Inet6Address/serialize/Inet6AddressSerializationTest.java [hunk 1]
```diff
@@ -177,12 +183,19 @@ static void testAllNetworkInterfaces() throws Exception {
...
             NetworkInterface netIF = e.nextElement();
             // Skip (Windows)Teredo Tunneling Pseudo-Interface
+            String dName = netIF.getDisplayName();
             if (isWindows) {
-                String dName = netIF.getDisplayName();
                 if (dName != null && dName.contains("Teredo")) {
                     continue;
                 }
             }
+            // skipp awdl and llw interfaces on macosx
```
Comment (jaikiran) [inline test/jdk/java/net/Inet6Address/serialize/Inet6AddressSerializationTest.java:192]: Hello Mark, this and a few other places have a typo for the word "skip"

**Review by jaikiran — APPROVED** [review]
The change looks OK to me. Please update the copyright year on the test file.

**Review by dfuch — COMMENTED** [review]
Thanks for this fix @msheppar !

**Pull request closed without merging**